[inductor][pattern matcher] Retain meta tags

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #102522
* #102450
* #102449
* #102578
* __->__ #102462
* #102448
* #102557
* #102547
* #102521

This will be used later on while propagating the `recompute` flag all the way to min-cut partitioner.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

anijain2305 committed:
[inductor][pattern matcher] Retain meta tags

[ghstack-poisoned]

jansel commented on torch/_inductor/pattern_matcher.py:
```diff
@@ -603,13 +606,13 @@ def run_node(self, node) -> Any:
                         result.meta["val"] = node.meta["val"]
+                    # Retain the meta tags from the first node in the match.
+                    # This is useful for retaining tags like recompute.
+                    result.meta.update(first_node.meta)
```
Do we need to skip updating meta["val"] here?

jansel commented on torch/_inductor/pattern_matcher.py:
```diff
@@ -603,13 +606,13 @@ def run_node(self, node) -> Any:
-        node = output_nodes[0]
-
-        with graph.inserting_before(node):
+        with graph.inserting_before(first_node):
```
Is this change needed?

anijain2305 replied:
Oh, this is just variable name change.

anijain2305 committed:
Update on "[inductor][pattern matcher] Retain meta tags"


This will be used later on while propagating the `recompute` flag all the way to min-cut partitioner.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

anijain2305 committed:
Update on "[inductor][pattern matcher] Retain meta tags"


This will be used later on while propagating the `recompute` flag all the way to min-cut partitioner.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

anijain2305 committed:
Update on "[inductor][pattern matcher] Retain meta tags"


This will be used later on while propagating the `recompute` flag all the way to min-cut partitioner.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

anijain2305 committed:
Update on "[inductor][pattern matcher] Retain meta tags"


This will be used later on while propagating the `recompute` flag all the way to min-cut partitioner.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

anijain2305 committed:
Update on "[inductor][pattern matcher] Retain meta tags"


This will be used later on while propagating the `recompute` flag all the way to min-cut partitioner.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

anijain2305 committed:
Update on "[inductor][pattern matcher] Retain meta tags"


This will be used later on while propagating the `recompute` flag all the way to min-cut partitioner.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

anijain2305 committed:
Update on "[inductor][pattern matcher] Retain meta tags"


This will be used later on while propagating the `recompute` flag all the way to min-cut partitioner.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot merge
